Userland/Games/BrickGame/BrickGame.cpp

```diff
@@ -0,0 +1,554 @@
+        }
+        auto const now { Time::now_realtime() };
+        auto const delay = s_level_map[m_level].m_delay;
+        if (now - m_last_update > delay) {
+            m_last_update = now;
+            move_down();
+        }
+    }
+
+    void reset()
+    {
+        m_level = 0;
+        m_score = 0;
+        m_well.reset();
+        m_is_game_over = false;
+        m_last_update = Time::now_realtime();
+        m_block.random_shape();
+        m_next_block.random_shape();
+    }
+
+private:
+    Well m_well {};
+    Block m_block {};
+    Block m_next_block {};
+    unsigned m_level {};
+    unsigned m_score {};
+    bool m_is_game_over {};
+    Time m_last_update {};
+
+    struct LevelMap final {
+        unsigned const m_score;
+        Time const m_delay;
+    };
+
+    static constexpr Array<LevelMap, 14> s_level_map = {
+        LevelMap { 0, Time::from_milliseconds(38000 / 60) },
+        LevelMap { 1000, Time::from_milliseconds(34000 / 60) },
+        LevelMap { 2000, Time::from_milliseconds(29000 / 60) },
+        LevelMap { 3000, Time::from_milliseconds(25000 / 60) },
+        LevelMap { 4000, Time::from_milliseconds(22000 / 60) },
+        LevelMap { 5000, Time::from_milliseconds(18000 / 60) },
+        LevelMap { 6000, Time::from_milliseconds(15000 / 60) },
+        LevelMap { 7000, Time::from_milliseconds(11000 / 60) },
+        LevelMap { 8000, Time::from_milliseconds(7000 / 60) },
+        LevelMap { 9000, Time::from_milliseconds(5000 / 60) },
+        LevelMap { 10000, Time::from_milliseconds(4000 / 60) },
+        LevelMap { 20000, Time::from_milliseconds(3000 / 60) },
+        LevelMap { 30000, Time::from_milliseconds(2000 / 60) },
+        LevelMap { 10000000, Time::from_milliseconds(1000 / 60) }
+    };
+
+    void set_current_block(Block const& block)
+    {
+        if (!block.has_collision(m_well))
+            m_block = block;
+    }
+
+    void check_and_remove_full_rows()
+    {
+        auto const num_removed_rows { m_well.check_and_remove_full_rows() };
+        switch (num_removed_rows) {
+        case 1:
+            m_score += 40 * (m_level + 1);
+            break;
+        case 2:
+            m_score += 100 * (m_level + 1);
+            break;
+        case 3:
+            m_score += 300 * (m_level + 1);
+            break;
+        case 4:
+            m_score += 1200 * (m_level + 1);
+            break;
+        default:
+            break;
+        }
+    }
+};
+
+BrickGame::BrickGame(StringView app_name)
+    : m_app_name { app_name }
+    , m_state { GameState::Idle }
+    , m_brick_game(make<Bricks>())
+{
+    set_font(Gfx::FontDatabase::default_fixed_width_font().bold_variant());
+    m_high_score = Config::read_i32(m_app_name, m_app_name, "HighScore"sv, 0);
+    reset();
+}
+
+void BrickGame::reset()
+{
+    m_state = GameState::Active;
+    m_brick_game->reset();
+    stop_timer();
+    start_timer(15); // 66.6ms
+    if (!m_brick_game->new_block())
+        VERIFY_NOT_REACHED();
```

Comment:
I slightly mislead you here, because `new_block` returns a `bool`, it would make even more sense to say:
`VERIFY(m_brick_game->new_block())`

However, it isn't obvious the return value of `new_block`, and what that `bool` represents, without going through and looking at the function itself.
Would it be possible to combine `is_game_over` logic into the `GameState` enum?
The method `is_game_over` could even stay, and instead just check if the `GameState` is `Over`, or something akin to that... depends if you believe that implementation makes sense though!